proposals/0010-diet-clients.txt

```diff
@@ -0,0 +1,183 @@
+---
+simd: '0010'
+title: Diet Clients
+authors:
+  - Anatoly Yakovenko (Solana Labs)
+category: Sealevel Virtual Machine
+type: Standards Track
+status: Draft
+created: 2022-12-09
+---
+
+## Summary
+
+Users that don't run a full node need secure access to the solana
+blockchain. The goal of this proposal is to define a protocol for
+sampling the validators such that if at least a small minority of
+validators are not faulty the user can be notified of an invalid
+state transition or a double confirmation attack on the network.
+
+## Specification
+
+### Overview
+
+1. User observes a confirmation for a TX at slot N. Confirmation
+contains 2/3+ votes for slot N, and a path from the transaction to
+the bankhash signed by all the votes.
+
+2. Concurrently with 1, user requests votes for slot N from M random
+validators.  Votes returned may be for later slots, and would contain
+a merkle path to the requested slot's bankhash.
+
+3. If all the responses are the same, the user is done. 
+
+4. If X% or more of staked nodes respond that they do not have block,
+the user starts sampling nodes at random for missing shreds and
+submitting the shreds to the validators with the missing block.
+
+    a. This process may need to run recursively until the fork block
+    is repaired.
+
+    b. If shreds cannot be recovered, a fault has occurred. User
+    shouldn't trust the confirmation.
+
+5. If X% or more of the staked nodes respond with different
+bankhash, a fault has occurred on the network. User shouldn't
+trust the confirmation. User should then submit the conflicting
+votes to M random validators.
```

Comment:
If the user observed a confirmation (2/3+ of votes) and is not trusting it, does the user in this case conclude that there are more than 1/3+ byzantine nodes?